src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -53,10 +53,16 @@ public static AuthenticationBuilder AddScriptJwtBearer(this AuthenticationBuilde
                     },
                     OnTokenValidated = c =>
                     {
-                        c.Principal.AddIdentity(new ClaimsIdentity(new Claim[]
+                        var claims = new List<Claim>
                         {
                             new Claim(SecurityConstants.AuthLevelClaimType, AuthorizationLevel.Admin.ToString())
-                        }));
+                        };
+                        if (string.Compare(c.SecurityToken.Issuer, ScriptConstants.AppServiceCoreUri) != 0)
```

Comment:
Can you use `string.Equals` here instead of and make sure we also pass a `StringComparison` to be explicit about the casing/type?